qiskit/circuit/library/standard_gates/sx.py

```diff
@@ -266,3 +272,6 @@ def _define(self):
         for operation, qubits, clbits in rules:
             qc._append(operation, qubits, clbits)
         self.definition = qc
+
+    def __eq__(self, other):
+        return isinstance(other, CSXGate)
```

Comment:
We've dropped the `ctrl_state` check from this one.

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/11370/commits/069184fddd7a4343b77803c02160d1ce104a8897